v20.2/build-a-nodejs-app-with-cockroachdb.md

```diff
@@ -1,104 +1,215 @@
-- Replace the value for `password` with the password you created for your user.
-- Replace the value for `host` with the name of the CockroachCloud Free host (e.g., `host: 'free-tier.gcp-us-central1.cockroachlabs.cloud'`).
-- Replace the value for `port` with the port to your cluster.
-- Replace the value for `database` with the database that you created earlier, suffixed with the name of the cluster (e.g., `database: '{cluster_name}.bank'`).
-- Remove the existing `ssl` object and its contents.
-- Uncomment the `ssl` object with the `ca` key-value pair, and edit the `fs.readFileSync('/certs/ca.crt').toString()` call to use the path to the `cc-ca.crt` file that you downloaded from the CockroachCloud Console.
+    You'll use this connection string to connect to the database later in this tutorial.
 
 </section>
 
-## Step 6. Run the code
+## Step 2. Get the code
 
-The sample code creates a table, inserts some rows, and then reads and updates values as an atomic [transaction](transactions.html).
+Clone the code's GitHub repo:
 
-Here are the contents of `app.js`:
+{% include_cached copy-clipboard.html %}
+~~~ shell
+$ git clone https://github.com/cockroachlabs/example-app-node-postgres
+~~~
+
+The project has the following directory structure:
+
+~~~
+├── README.md
+├── app.js
+├── dbinit.sql
+└── package.json
+~~~
+
+The `dbinit.sql` file initializes the database schema that the application uses:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+{% remote_include https://raw.githubusercontent.com/cockroachlabs/example-app-node-postgres/main/dbinit.sql %}
+~~~
+
+The `app.js` file contains the code for `INSERT`, `SELECT`, `UPDATE`, and `DELETE` SQL operations:
 
 {% include_cached copy-clipboard.html %}
 ~~~ js
 {% remote_include https://raw.githubusercontent.com/cockroachlabs/example-app-node-postgres/main/app.js %}
 ~~~
 
-Note that all of the database operations are wrapped in the `retryTxn` function. This function attempts to commit statements in the context of an explicit transaction. If a [retry error](transaction-retry-error-reference.html) is thrown, the wrapper will retry committing the transaction, with [exponential backoff](https://en.wikipedia.org/wiki/Exponential_backoff), until the maximum number of retries is reached (by default, 15).
+All of the database operations are wrapped in a helper function named `retryTxn`. This function attempts to commit statements in the context of an explicit transaction. If a [retry error](transaction-retry-error-reference.html) is thrown, the wrapper will retry committing the transaction, with [exponential backoff](https://en.wikipedia.org/wiki/Exponential_backoff), until the maximum number of retries is reached (by default, 15).
+
+## Step 3. Initialize the database
+
+To initialize the example database, pass the `dbinit.sql` file to the [`cockroach sql`](cockroach-sql.html) command:
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+cockroach sql --url '<connection-string>' < dbinit.sql
```

Comment:
The point being, if you wanted to move the command back inside the app, you could do so by using `cat`.

Reply:
Thanks for digging this up! I've updated the command, as you suggested. I made it an include, just in case we want to do this for other apps down the line.

> if you wanted to move the command back inside the app, you could do so by using `cat`

I'm OK with keeping it out for this one. Makes the application code less cluttered.

Reply:
Ok, sounds like a plan.